c7n/resources/datasync.py

```diff
@@ -0,0 +1,38 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+from c7n.manager import resources
+from c7n import query
+
+
+@resources.register("datasync-agent")
+class DataSyncAgent(query.QueryResourceManager):
+    class resource_type(query.TypeInfo):
+        service = "datasync"
+        enum_spec = ('list_agents', 'Agents', None)
+        detail_spec = ('describe_agent', 'AgentArn', 'AgentArn', '')
+        arn_type = "agent"
+        arn = "AgentArn"
+        id = "AgentArn"
+        name = "Name"
+        cfn_type = "AWS::DataSync::Agent"
+        universal_taggable = object()
+
+    source_mapping = {
+        "describe": query.DescribeWithResourceTags,
+    }
+
+
+@resources.register("datasync-task")
+class DataSyncTask(query.QueryResourceManager):
+    class resource_type(query.TypeInfo):
+        service = "datasync"
+        enum_spec = ('list_tasks', 'Tasks', None)
+        detail_spec = ('describe_task', 'TaskArn', 'TaskArn', '')
```

Comment:
This also pulls down all the `unavailable` tasks. Might be worth only looking at tasks with `"Status": "AVAILABLE"`? Not sure how this service works but in the long run there could be thousands of these tasks potentially in UNAVAILABLE state

Reply:
https://docs.aws.amazon.com/datasync/latest/userguide/understand-task-statuses.html

an unavailable task state seems like something that would be of interest (agent is down, system not working as expected). I'm okay with adding as is for now, and going back to add a default server side query later if there's interest.